packages/tx-construction/src/tx-builder/TokenBuilder.ts

```diff
@@ -0,0 +1,26 @@
+import { Cardano } from '@cardano-sdk/core';
+
+export class TxTokenBuilder {
+  #assets: Cardano.TokenMap;
+  #policyId: Cardano.PolicyId;
+
+  constructor(policyId: Cardano.PolicyId) {
+    this.#policyId = policyId;
+    this.#assets = new Map();
+  }
+
+  #toHex = (value: string) =>
+    value
+      .split('')
+      .map((s) => s.charCodeAt(0).toString(16))
+      .join('');
+
+  addAsset(tokenName: string, quantity: bigint): TxTokenBuilder {
+    this.#assets.set(Cardano.AssetId(`${this.#policyId}${this.#toHex(tokenName)}`), quantity);
+    return this;
+  }
+
+  build(): Cardano.TokenMap {
+    return this.#assets ? this.#assets : new Map();
```

Comment:
this.#assets is initialised to `new Map()` in constructor.
Also, please return a copy of the assets. Otherwise, the returned assets map is just a reference to the internal `#assets` of the TxTokenBuilder and updating it from the TxTokenBuilder will also update the previously returned `#assets`.
```
-     return this.#assets ? this.#assets : new Map();
+    return new Map(this.#assets);;